### DISCUSSION TITLE
iOS clearing fields on reruns of failed tests

### DISCUSSION THREAD

**Discussion by lefty1975** [appium/appium#17381] [General]
I’m trying to see if anyone has ever encountered where iOS does not clear input fields on reruns of a failed test. I’m assuming this occurs because Appium qualifies this as the same session and is retaining data, but I would like to see if there’s a capability that will work. I have tried the noReset and fullReset options to no avail and I would rather not write logic to handle this relatively dynamic state.

**Comment by RSM-SBA** [comment]
i write small tests and do a restart of the app (with or without reseting it) to avoid such situations.
I think appium does not have an inbuild logic that resets input fields if they are filled.
Otherwise you can always do a .clear() on input fields before doing a .sendKeys()

**Reply by lefty1975** [reply]
Thank you for the feedback.

**Comment by jlipps** [comment]
Appium neither retains data nor does anything special to clear it out. If you want data to be cleared out on test reruns, make sure your reruns are in a different session and that you're not using the `appium:noReset` capability.
